### PR TITLE
This code is working with Beta Version of Kubebuilder.

### PR DESCRIPTION
**Pull request by laxmikantbpandhare**
This code is working with Beta Version of Kubebuilder. Please review and approve.

### CODE REVIEW

File: pkg/java/v1/api.go [hunk 1]
```diff
@@ -17,34 +17,47 @@
...
 	"sigs.k8s.io/kubebuilder/v3/pkg/plugin"
+	pluginutil "sigs.k8s.io/kubebuilder/v3/pkg/plugin/util"
 
-	"github.com/java-operator-sdk/kubebuilder-plugin/pkg/internal/kubebuilder/cmdutil"
 	"github.com/java-operator-sdk/kubebuilder-plugin/pkg/java/v1/scaffolds"
 )
 
+type createAPIOptions struct {
+	CRDVersion         string
+	DoRole, DoPlaybook bool
```
Comment (jmrodri) [inline pkg/java/v1/api.go]: These are ansible specific. Remove them.
```suggestion
```

File: pkg/java/v1/scaffolds/api.go [hunk 2]
```diff
@@ -2,18 +2,31 @@ package scaffolds
...
-func NewCreateAPIScaffolder() cmdutil.Scaffolder {
-	return &apiScaffolder{}
+// NewCreateAPIScaffolder returns a new plugins.Scaffolder for project initialization operations
+func NewCreateAPIScaffolder(cfg config.Config, res resource.Resource, doRole, doPlaybook bool) plugins.Scaffolder {
+	return &apiScaffolder{
+		config:     cfg,
+		resource:   res,
+		doRole:     doRole,
+		doPlaybook: doPlaybook,
+	}
```
Comment (jmrodri) [inline pkg/java/v1/scaffolds/api.go]: role and playbook are ansible specific, remove them.
```suggestion
func NewCreateAPIScaffolder(cfg config.Config, res resource.Resource) plugins.Scaffolder {
	return &apiScaffolder{
		config:     cfg,
		resource:   res,
	}
```

File: pkg/java/v1/scaffolds/api.go [hunk 3]
```diff
@@ -2,18 +2,31 @@ package scaffolds
...
-	"github.com/java-operator-sdk/kubebuilder-plugin/pkg/internal/kubebuilder/cmdutil"
 )
 
 type apiScaffolder struct {
 	fs machinery.Filesystem
+
+	config   config.Config
+	resource resource.Resource
+
+	doRole, doPlaybook bool
```
Comment (jmrodri) [inline pkg/java/v1/scaffolds/api.go]:
```suggestion

	config   config.Config
	resource resource.Resource
```

**Review by jmrodri — CHANGES_REQUESTED** [review]
Remove the role and playbook variables which are specific to ansible. The rest looks great.

File: pkg/java/v1/scaffolds/api.go (hunk 3 above)
Comment (laxmikantbpandhare) [reply pkg/java/v1/scaffolds/api.go]: Modified according to suggestion.

File: pkg/java/v1/scaffolds/api.go (hunk 2 above)
Comment (laxmikantbpandhare) [reply pkg/java/v1/scaffolds/api.go]: Modified according to suggestion.

File: pkg/java/v1/api.go (hunk 1 above)
Comment (laxmikantbpandhare) [reply pkg/java/v1/api.go]: Modified according to suggestion.

**Review by jmrodri — APPROVED** [review]

**Pull request merged**